pilot/pkg/xds/endpoint_builder.go

```diff
@@ -20,16 +20,38 @@ import (
 
 	core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
 	endpoint "github.com/envoyproxy/go-control-plane/envoy/config/endpoint/v3"
+	"github.com/golang/protobuf/proto"
 	"github.com/golang/protobuf/ptypes/wrappers"
 
 	networkingapi "istio.io/api/networking/v1alpha3"
 	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/networking"
 	"istio.io/istio/pilot/pkg/networking/util"
 	"istio.io/istio/pkg/config"
 	"istio.io/istio/pkg/config/host"
 	"istio.io/istio/pkg/config/schema/gvk"
 )
 
+// Return the tunnel type for this endpoint builder. If the endpoint builder builds h2tunnel, the final endpoint
+// collection includes only the endpoints which support H2 tunnel and the non-tunnel endpoints. The latter case is to
+// support multi-cluster service.
+// Revisit non-tunnel endpoint decision once the gateways supports tunnel.
+// TODO(lambdai): Propose to istio api.
+func GetTunnelBuilderType(clusterName string, proxy *model.Proxy, push *model.PushContext) networking.TunnelType {
+	if proxy == nil || proxy.Metadata == nil || proxy.Metadata.ProxyConfig == nil {
+		return networking.NoTunnel
+	}
+	if outTunnel, ok := proxy.Metadata.ProxyConfig.ProxyMetadata["tunnel"]; ok {
```

Comment:
You should add Tunnel as a field to `proxy.Metadata`. `ProxyMetadata` is just one way to configure the node metadata and probably shouldn't have been passed at all (but we pass Proxyconfig which contains it).

Reply:
So the next step is to 
1. declare api-ish tunnel in ProxyMetadata 
2. populate the field tunnel by `proxy.Metadata`, or other viable source
Update this helper method and the test